nengo/builder/optimizer.py

```diff
@@ -0,0 +1,974 @@
+            del by_view[None]
+
+        for view_subset in by_view.values():
+            if len(view_subset) > 1:
+                self._perform_merges_for_view_subset(
+                    view_subset, tc, op_replacements, sig_replacements)
+
+        return op_replacements, sig_replacements
+
+    def _perform_merges_for_view_subset(
+            self, subset, tc, op_replacements, sig_replacements):
+        """Perform merges for a subset of operators with the same view base.
+
+        Parameters
+        ----------
+        subset : list
+            Subset of operators. These need to have the same view base (can be
+            None if it is None for all) for their first signal in
+            `all_signals`.
+        tc : dict
+            Transitive closure of dependency graph.
+        only_merge_ops_with_view : bool
+            Limit merges to operators with views.
+        op_replacements : dict
+            Dictionary that maps from original operators to the newly merged
+            operators. To be used to update the dependency graph. This dict
+            will be updated by this function.
+        sig_replacements : dict
+            Dictionary that maps from original signals to new signals resulting
+            from the merges. To be used to update the signals referenced by
+            the operators. This dict will be updated by this function.
+        """
+        view_indices = []
+        # Sort to have sequential memory.
+        offsets = np.array([self._first_view_offset(op) for op in subset])
+        sort_indices = np.argsort(offsets)
+        offsets = offsets[sort_indices]
+        sorted_subset = [subset[i] for i in sort_indices]
+
+        for i, op1 in enumerate(sorted_subset):
+            if op1 in self._merged:
+                # Cannot merge merged operator again until dependency graph
+                # has been updated (happens at the end of one optimization
+                # pass by calling _update_dg).
+                continue
+
+            view_indices = self._get_view_indices(op1)
+
+            # Find operators that can be merged.
+            merge = [op1]
+            # For a merge to be possible the view of the next operator has to
+            # start where the view of op1 ends. Because we have sorted the
+            # operators by the start of their views we can do a binary search
+            # and potentially skip a number of operators at the beginning.
+            start = np.searchsorted(
+                offsets, offsets[i] + self._first_view_size(op1), side='left')
+            for op2 in sorted_subset[start:]:
+                can_merge = (
+                    op2 not in self._merged and
+                    self._mergeable(merge[-1], op2, view_indices, tc) and
+                    self._is_memory_access_sequential([merge[-1], op2]))
+                if can_merge:
+                    merge.append(op2)
+                elif self._check_sequential(merge[-1], op2):
```

Comment:
Ah okay. So is it possible that this check will be True, but the previous check will be True also? In other words, could this be the if instead of the elif?